### ISSUE TITLE
Forwarding: immediately processing vote on target server

### ISSUE THREAD

**Issue by x1p** [NuVotifier/NuVotifier#155]
When the forward method is set to pluginMessaging, the votifier on the spigot server does not process the vote unless someone is online.
Is it possible to have this processed even if there is no one online?

Global idea how the setup is:
Bungee:
```
forwarding:
  method: pluginMessaging
    onlySendToJoinedServer: false
```
The spigot version of votifier lacks the possibility `onlySendToJoinedServer: false`

When set to `proxy` the vote is processed immediately by the spigot server whether someone is online or not. This way the listener (like VotingPlugin) can decide to handle to whatever I set it to.

So I could set it to proxy. But proxy lacks caching. So if, for whatever reason, the Hub should restart, votes might get lost. Since there are no means of setting it to retry a little longer. It skips the vote after only 15 sec or so. A normal restart of a Hub takes much longer.

**Comment by Ichbinjoe** [comment]
No, this is not possible. This is a deficiency with how plugin messages are implemented - they piggyback on top of online players.

The idea is that most vote rewards for simple servers with no players on it don't need the vote _right then_, but can wait until one player gets on. Then we usually rely on vote listeners on the server to do the application specific storage because NV doesn't have the context (beyond routing) to understand what you actually need the vote for, so it does best effort.

onlySendToJoinedServer is only on the proxy layer because it only informs the routing decision - beyond routing it has no effect.

**Comment by Ichbinjoe** [comment]
Proxy mode / plugin messaging forwarding is going away in 3.0.0. Keeping open for the time being but the solution will come in a different form.

**Comment by Kaktuz1283** [comment]
Hey! I would like to know how can I setup my Votifier correctly in Bungee and spigot servers.
I'm using Votifier and Voting Plugin (rewards) and I have one problem: when users vote for the network, the reward goes to all servers, and I would like to send it just to the joined server.

I hope this is useful to know the mistake I'm making:

forwarding:
  method: proxy
  pluginMessaging:
    channel: nuvotifier:votes
    excludedServers:
    cache: file
    onlySendToJoinedServer: true
    joinedServerFallback: 'survival'

I don't know if I have to change the method to "pluginMessaging", English is not my native language and the instructions in config are a little bit confusing for the feature that I want.
Thank you!

**Comment by astei** [comment]
@Kaktuz1283 This is a bug tracker, not a support channel.

**Comment by Kaktuz1283** [comment]
@astei Where can I get the support then, please?

**Comment by astei** [comment]
Ask whoever made your vote listener.

**Comment by Kaktuz1283** [comment]
So it isn't a problem of Votifier? I don't really understand

**Comment by astei** [comment]
[If you'd read the Spigot page you'd know](https://www.spigotmc.org/resources/nuvotifier.13449/). I am no longer going to waste my time hand-holding people regarding setup problems.

**Comment by Kaktuz1283** [comment]
You're right! I didn't read that correctly. Thanks for your kind and timely answers! 😆
